library/src/main/kotlin/wu/seal/jsontokotlin/JsonToKotlinBuilder.kt

```diff
@@ -0,0 +1,268 @@
+            isNestedClassModel = false
+            enableMinimalAnnotation = false
+            indent = 4
+            parenClassTemplate = ""
+            isKeywordPropertyValid = true
+            extensionsConfig = ""
+        }
+    }
+
+    /**
+     * To set property type to `var`, pass true.
+     * Default : false
+     */
+    fun enableVarProperties(isVar: Boolean): JsonToKotlinBuilder {
+        TestConfig.isPropertiesVar = isVar
+        return this
+    }
+
+    /**
+     * To set if the properties can be null or not
+     * Default: PropertyTypeStrategy.NotNullable
+     */
+    fun setPropertyTypeStrategy(strategy: PropertyTypeStrategy): JsonToKotlinBuilder {
+        TestConfig.propertyTypeStrategy = strategy
+        return this
+    }
+
+    /**
+     * To set default value.
+     * Default : DefaultValueStrategy.AvoidNull
+     */
+    fun setDefaultValueStrategy(strategy: DefaultValueStrategy): JsonToKotlinBuilder {
+        TestConfig.defaultValueStrategy = strategy
+        return this
+    }
+
+    /**
+     * To set JSON decoding/encoding library
+     * Default: TargetJsonConverter.None
+     */
+    fun setAnnotationLib(library: TargetJsonConverter): JsonToKotlinBuilder {
+
+        require(library != TargetJsonConverter.Custom) { "Call #setCustomAnnotation to set custom annotation" }
+
+        TestConfig.targetJsonConvertLib = library
+        return this
+    }
+
+    fun setCustomAnnotation(
+            importClass: String,
+            classAnnotationFormat: String,
+            propertyAnnotationFormat: String
+    ): JsonToKotlinBuilder {
+        TestConfig.apply {
+            targetJsonConvertLib = TargetJsonConverter.Custom
+            customAnnotaionImportClassString = importClass
+            customClassAnnotationFormatString = classAnnotationFormat
+            customPropertyAnnotationFormatString = propertyAnnotationFormat
+        }
+
+        return this
+    }
+
+    /**
+     * If enabled, value will be commented right to the property
+     */
+    fun enableComments(isEnable: Boolean): JsonToKotlinBuilder {
+        TestConfig.isCommentOff = !isEnable
+        return this
+    }
+
+
+    /**
+     * TODO: More comments need to be added
+     */
+    fun enableOrderByAlphabetic(isOrderByAlphabetic: Boolean): JsonToKotlinBuilder {
+        TestConfig.isOrderByAlphabetical = isOrderByAlphabetic
+        return this
+    }
+
+
+    fun enableInnerClassModel(isInnerClassModel: Boolean): JsonToKotlinBuilder {
+        TestConfig.isNestedClassModel = isInnerClassModel
+        return this
+    }
+
+    fun enabelMapType(isMapType: Boolean): JsonToKotlinBuilder {
+        TestConfig.enableMapType = isMapType
+        return this
+    }
+
+    fun enableCreateAnnotationOnlyWhenNeeded(isOnlyWhenNeeded: Boolean): JsonToKotlinBuilder {
+        TestConfig.enableMinimalAnnotation = isOnlyWhenNeeded
+        return this
+    }
+
+    /**
+     * To set indent in output kotlin code.
+     *
+     * Default : 4
```

Comment:
currently there only this method with test case doc? Do we need to add others?

Reply:
Yes we need to add doc to all methods, but for now, I hope it'll be better if we add docs to methods that are not self explainable. What do you think ?

Reply:
Yes, U are right